app/react/Attachments/components/AttachmentsModal.tsx

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable react/prop-types */
+
+import React, { useRef } from 'react';
+import { connect } from 'react-redux';
+import ReactModal from 'react-modal';
+import { Tabs, TabLink, TabContent } from 'react-tabs-redux';
+import Dropzone from 'react-dropzone';
+import { LocalForm, Field } from 'react-redux-form';
+
+import { Translate } from 'app/I18N';
+import { Icon } from 'app/UI';
+
+import { uploadAttachment, uploadAttachmentFromUrl } from '../actions/actions';
+
+const validators = {
+  name: { required: (val: any) => !!val && val.trim() !== '' },
+  url: { required: (val: any) => !!val && val.trim() !== '' },
+};
+
+interface AttachmentsModalProps {
+  isOpen: boolean;
+  entitySharedId: string;
+  entityId: string;
+  storeKey: string;
+  onClose(): void;
+  uploadAttachment(entity: any, file: any, __reducerKey: any, options?: {}): void;
+  uploadAttachmentFromUrl(entity: any, name: any, url: any, __reducerKey: any): void;
+  getPercentage?: number;
+}
+
+export const AttachmentsModalCmp: React.FC<AttachmentsModalProps> = ({
+  isOpen,
+  entitySharedId,
+  storeKey,
+  onClose,
+  uploadAttachment: uploadAttachmentProp,
+  uploadAttachmentFromUrl: uploadAttachmentFromUrlProp,
+  getPercentage,
+}) => {
```

Comment:
Please specify the type of the argument: 
`...
  getPercentage,
}: AttachmentsModalProps) => {`

Reply:
Type is specified in `AttachmentsModalProps`. It's `number | undefined`.